ios/FluentUI/Bottom Sheet/BottomSheetController.swift

```diff
@@ -491,57 +511,93 @@ public class BottomSheetController: UIViewController {
-                        self.expandedContentView.alpha = targetExpandedContentAlpha
-                    }
-                }
+    private func stateChangeAnimator(to targetExpansionState: BottomSheetExpansionState,
+                                     velocity: CGFloat = 0.0,
+                                     interaction: BottomSheetInteraction = .noUserAction,
+                                     shouldNotifyDelegate: Bool = true) -> UIViewPropertyAnimator {
+        let targetOffsetFromBottom = offset(for: targetExpansionState)
+        let distanceToGo = abs(currentOffsetFromBottom - targetOffsetFromBottom)
+        let springVelocity = min(abs(velocity / distanceToGo), Constants.Spring.maxInitialVelocity)
+        let damping: CGFloat = abs(velocity) > Constants.Spring.flickVelocityThreshold
+            ? Constants.Spring.oscillatingDampingRatio
+            : Constants.Spring.defaultDampingRatio
 
-                if shouldShowDimmingView {
-                    let targetDimmingViewAlpha: CGFloat = targetExpansionState == .expanded ? 1.0 : 0.0
-                    translationAnimator.addAnimations {
-                        self.dimmingView.alpha = targetDimmingViewAlpha
-                    }
-                }
+        let springParams = UISpringTimingParameters(dampingRatio: damping, initialVelocity: CGVector(dx: 0.0, dy: springVelocity))
+        let translationAnimator = UIViewPropertyAnimator(duration: Constants.Spring.animationDuration, timingParameters: springParams)
 
-                translationAnimator.addCompletion({ [weak self] finalPosition in
-                    if finalPosition == .end {
-                        self?.handleCompletedStateChange(to: targetExpansionState, interaction: interaction, shouldNotifyDelegate: shouldNotifyDelegate)
-                    }
-                    completion?(finalPosition)
-                })
-                translationAnimator.startAnimation()
-            } else {
-                bottomSheetOffsetConstraint.constant = -targetOffsetFromBottom
-                handleCompletedStateChange(to: targetExpansionState, interaction: interaction, shouldNotifyDelegate: shouldNotifyDelegate)
-                completion?(.end)
+        // Disable dragging while hiding the sheet
+        if targetExpansionState == .hidden {
+            panGestureRecognizer.isEnabled = false
+        }
+
+        view.layoutIfNeeded()
+
+        // Animation might be reversed, so we need to remember the original state
+        let originalBottomSheetHiddenState = bottomSheetView.isHidden
+        let originalBottomOffsetConstant = bottomSheetOffsetConstraint.constant
+
+        bottomSheetView.isHidden = false
+        bottomSheetOffsetConstraint.constant = -targetOffsetFromBottom
+        translationAnimator.addAnimations { [weak self] in
+            self?.view.layoutIfNeeded()
+        }
+
+        let targetExpandedContentAlpha: CGFloat = targetExpansionState == .collapsed ? 0.0 : 1.0
```

Comment:
What about creating computed properties for targetExpandedContentAlpha and targetDimmingViewAlpha in the BottomSheetExpansionState to return these values instead?

Reply:
I like that! I also changed the logic for expanded content alpha because I realized that alpha should really only be 1.0 in .expanded state.

Reply:
Will there ever be a situation where they differ?
I'm asking because their implementation is the same now:

var expandedContentAlpha: CGFloat **{ self == .expanded ? 1.0 : 0.0 }**
var dimmingViewAlpha: CGFloat **{ self == .expanded ? 1.0 : 0.0 }**

Would that be a good idea to have just one property like "viewAlphaValue" and apply on both scenarios since you now changed its logic?

Reply:
Yeah it's kind of a happy coincidence they end up with the same alpha, but they're unrelated. If we add another expansion state (like a midpoint), they will diverge. I do agree with merging them for now tho.